app/src/main/java/org/mozilla/fenix/telemetry/TabsTrayMiddleware.kt

```diff
@@ -0,0 +1,65 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.telemetry
+
+import androidx.annotation.VisibleForTesting
+import mozilla.components.lib.state.Middleware
+import mozilla.components.lib.state.MiddlewareContext
+import org.mozilla.fenix.components.metrics.Event
+import org.mozilla.fenix.components.metrics.MetricController
+import org.mozilla.fenix.tabstray.TabsTrayAction
+import org.mozilla.fenix.tabstray.TabsTrayState
+import org.mozilla.fenix.utils.Settings
+
+/**
+ * [Middleware] that reacts to various [TabsTrayAction]s.
+ *
+ * @property settings reference to the application [Settings].
+ * @property metrics reference to the configured [MetricController] to record general page load events.
+ */
+class TabsTrayMiddleware(
```

Comment:
This is my first middleware, so hoping this is what @jonalmeida had in mind with the todo in TabSorter